pkg/models/host.go

```diff
@@ -73,10 +59,16 @@ func (h *Host) GetCloudID() string {
 }
 
 // Connect starts a new SSH connection with the provided private key.
-func (h *Host) Connect(timeout time.Duration) error {
-	h.Connection = NewHostConnection(*h, timeout)
-	if !h.Connected() {
-		return fmt.Errorf("failed to connect to host %s", h.IP)
+func (h *Host) Connect() error {
+	if h.Connection != nil {
+		return nil
+	}
+	var err error
+	for i := 0; h.Connection == nil && i < sshConnectionRetries; i++ {
+		h.Connection, err = NewHostConnection(h)
```

Comment:
we need to put sleep on exponensial sleep in between

Reply:
No need for sleep. Connection attempt already takes a lot of time and this works pretty fine.